### PR TITLE
Tbl087/adding more secure message endpoints no error codes

### PR DESCRIPTION
**Pull request by andrewmil**
Update on this PR https://github.com/ONSdigital/ras-frontstage-api/pull/6
Adding the remaining API endpoints required for the secure-messaging view in ras-frontstage

- Now endpoints at /messages_list, /message and /send_message
- Calls are now made to the case and party services so additional config needed for these
- Request handler has been simplified to only catch ConnectionError and return an error message

This PR works in conjunction with frontstage changes at https://github.com/ONSdigital/ras-frontstage/pull/229

### CODE REVIEW

File: frontstage_api/error_handlers.py [hunk 1]
```diff
@@ -1,25 +1,30 @@
...
 
-@api.errorhandler(FailedRequest)
-def failed_request(error):
-    message_json = {
-        'message': 'External request failed',
-        'url': error.url,
-        'method': error.method,
-        'exception': error.exception
+@app.errorhandler(ApiError)
+@api.errorhandler(ApiError)
```
Comment (benjefferies) [inline frontstage_api/error_handlers.py]: Would http://flask.pocoo.org/docs/0.12/api/#flask.Blueprint.app_errorhandler cover both the scenarios?

Comment (andrewmil) [reply frontstage_api/error_handlers.py]: `api.errorhandler` is the intended solution but I think there's a problem with it in Flask-Restplus, I'll try just `app.errorhandler` and see if it has the same effect

Comment (andrewmil) [reply frontstage_api/error_handlers.py]: yeah that doesn't work, I believe it's a known issue in restplus currently there's a couple of github issues about it. This is where I found this solution
https://github.com/noirbizarre/flask-restplus/issues/334

File: frontstage_api/controllers/secure_messaging_controllers.py [hunk 2]
```diff
@@ -89,41 +88,31 @@ def remove_unread_label(encoded_jwt, message_id):
...
-    if response.status_code != 200:
+    if not response or response.status_code != 200:
         logger.error('Error removing unread message label', status_code=response.status_code, message_id=message_id)
-        error_json = {
-            "error": {
-                "code": "FA005"
-            }
-        }
-        return error_json
+        unread_message_removed = False
```
Comment (tej99) [inline frontstage_api/controllers/secure_messaging_controllers.py]: unread_label_removed may be a better name here

File: frontstage_api/controllers/secure_messaging_controllers.py [hunk 3]
```diff
@@ -89,41 +88,31 @@ def remove_unread_label(encoded_jwt, message_id):
...
+    else:
+        logger.debug('Successfully removed unread label')
+        unread_message_removed = True
 
-    logger.debug('Successfully removed unread label')
-    return {"unread_message_removed": True}
+    return {"unread_message_removed": unread_message_removed}
```
Comment (tej99) [inline frontstage_api/controllers/secure_messaging_controllers.py]: same with the dict key returned

Comment (andrewmil) [reply frontstage_api/controllers/secure_messaging_controllers.py]: Will add

**Pull request merged**